### DISCUSSION TITLE
Convert existing js to ts?

### DISCUSSION THREAD

**Discussion by goforu** [quasarframework/quasar#10737] [Ideas / Proposals]
I have an existing quasar project coding in javascript. To take advantage of Typescript feature, I want to add typescript to my existing project. Unfortunately, I have tons of vue files to rewrite. Is there a better way to do that? For say like use both js and ts in project or a batch ts converter?

**Comment by ontwikkelfabriek** [comment]
@goforu 

Qv1(Vue2) or Qv2(Vue3)?

**Reply by goforu** [reply]
Qv2(Vue3)

**Comment by ontwikkelfabriek** [comment]
@goforu 

I did find this article how to convert a Vue3 CLI app from JS to TS ( See the **Add TypeScript Support** part):
https://www.thisdot.co/blog/your-first-vue-3-app-using-typescript

And this repo and article ( a bit older for Vue2):
https://johnpapa.net/vue-typescript/
https://github.com/webtoolsforall/vue-ts-converter

**Reply by goforu** [reply]
THANKS for your answer. Very appreciated. But I found the article may not handle my case. For one thing, the article's adding TS support is targeting Vue 2 which is class-based. When it comes to Vue 3 it has refactored by pure hand. For another, the converting from js to ts only applies for files with certain file extension, etc js. While ts inside vue files remains untouched.

**Reply by ontwikkelfabriek** [reply]
@goforu 

If you find something better than convert by hand, let us know!
